libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SmartStoreUpgradeManager.java

```diff
@@ -77,10 +78,12 @@ protected synchronized void upgradeSmartStore() {
 
         // If the installed version < v8.2.0, we need to migrate encryption keys.
         try {
-            final String majorVersionNum = installedVersion.substring(0, 3);
-            double installedVerDouble = Double.parseDouble(majorVersionNum);
+            final SdkVersion installedVersion = SdkVersion.parseFromString(installedVersionStr);
```

Comment:
Same question as above, what's the point of that try-catch? We are not doing anything with the calculated installedVersion.